source/hp/dof_handler.cc

```diff
@@ -599,12 +599,10 @@ namespace internal
         {
           const unsigned int dim = 1;
 
-          typedef DoFHandler<dim,spacedim> BaseClass;
-
           Assert (dof_handler.finite_elements != nullptr,
-                  typename BaseClass::ExcNoFESelected());
+                  typename DoFHandler<dim,spacedim>::ExcNoFESelected());
```

Comment:
Nope, not allowed -- the preprocessor will complain about there being 3 arguments to `Assert` separated by commas. You'll have to enclose the whole second argument in parentheses for this to work. (This is why the code introduces the `typedef` in the first place.)

Reply:
Thankfully adding some parentheses doesn't take too much effort :-)